src/object/cli_obj.c

```diff
@@ -2268,10 +2277,28 @@ shard_io(tse_task_t *task, struct shard_auxi_args *shard_auxi)
 static int
 shard_io_task(tse_task_t *task)
 {
-	struct shard_auxi_args		*shard_auxi;
+	struct shard_auxi_args	*shard_auxi;
+	daos_handle_t		 th;
+	int			 rc;
 
 	shard_auxi = tse_task_buf_embedded(task, sizeof(*shard_auxi));
 
+	/*
+	 * If this task belongs to a TX, and if the epoch we got earlier
+	 * doesn't contain a "chosen" TX epoch, then we may need to reinit the
+	 * task. (See dc_tx_get_epoch.) Because tse_task_reinit is less
+	 * practical in the middle of a task, we do it here at the beginning of
+	 * shard_io_task.
+	 */
+	th = shard_auxi->obj_auxi->th;
+	if (daos_handle_is_valid(th) && !dtx_epoch_chosen(&shard_auxi->epoch)) {
+		rc = dc_tx_get_epoch(task, th, &shard_auxi->epoch);
+		if (rc == DC_TX_GE_REINIT)
+			return tse_task_reinit(task);
+		else if (rc < 0)
```

Comment:
"else" is unnecessary since above "if" branch have returned.

Reply:
I meant to write this as one "if ... else if ... else if ..." block for checking rc. I think breaking it into two "if" blocks would not make things a bit worse. And there's no compelling reason to break it, is there?